stochy_data_mod.F90

```diff
@@ -1,115 +1,128 @@
- subroutine init_stochdata(domain,mype,nlevs,delt,iret)
+  subroutine init_stochdata(domain,mype,nlevs,delt,iret)
 !\callgraph
 
 ! initialize random patterns.  
-   use netcdf
-   implicit none
+    use netcdf
+    implicit none
 
-   type(domain_type),intent(inout):: domain
-   integer, intent(in) :: mype,nlevs
-   real(kind_phys), intent(in) :: delt
-   integer, intent(out) :: iret
-   real :: ones(6)
+    type(domain_type),intent(inout):: domain
+    integer, intent(in) :: mype,nlevs
+    real(kind_phys), intent(in) :: delt
+    integer, intent(out) :: iret
+    real :: ones(6)
 
-   real :: rnn1
-   integer :: nn,k,nm,stochlun,ierr,n
-   integer :: locl,indev,indod,indlsod,indlsev
-   integer :: l,jbasev,jbasod
-   integer :: jcapin,varid1,varid2
-   real(kind_phys),allocatable :: noise_e(:,:),noise_o(:,:)
-   include 'function_indlsod'
-   include 'function_indlsev'
-   include 'netcdf.inc'
-   stochlun=99
-   levs=nlevs
+    real :: rnn1
+    integer :: nn,k,nm,stochlun,ierr,n
+    integer :: locl,indev,indod,indlsod,indlsev
+    integer :: l,jbasev,jbasod
+    integer :: jcapin,varid1,varid2
+    real(kind_phys),allocatable :: noise_e(:,:),noise_o(:,:)
+    include 'function_indlsod'
+    include 'function_indlsev'
+    include 'netcdf.inc'
+    stochlun=99
+    levs=nlevs
 
-   iret=0
+    iret=0
 ! read in namelist
 
-   call get_nml_rec (domain,mype,real(delt),iret)
+    call get_nml_rec (domain,mype,real(delt),iret)
   
-   if (iret/=0) return  ! need to make sure that non-zero irets are being trapped.
-   if ( (.NOT. do_sppt) .AND. (.NOT. do_shum) .AND. (.NOT. do_skeb)  .AND. (lndp_type==0) .AND. (.NOT. do_spp)) return
+    if (iret/=0) return  ! need to make sure that non-zero irets are being trapped.
+    if ( (.NOT. do_sppt) .AND. (.NOT. do_shum) .AND. (.NOT. do_skeb)  .AND. (lndp_type==0) .AND. (.NOT. do_spp)) return
+
+    call initialize_spectral(gis_stochy)
+
+    allocate(noise_e(len_trie_ls,2),noise_o(len_trio_ls,2))
 
-   call initialize_spectral(gis_stochy)
-   allocate(noise_e(len_trie_ls,2),noise_o(len_trio_ls,2))
 ! determine number of random patterns to be used for each scheme.
-   do n=1,size(sppt)
-     if (sppt(n) > 0) then
+    do n=1,size(sppt)
+      if (sppt(n) > 0) then
         nsppt=nsppt+1
-     else
+      else
         exit
-     endif
-   enddo
-   if (is_rootpe()) print *,'nsppt = ',nsppt
-   if (is_rootpe()) print *,'sppt_lscale = ',sppt_lscale
-   if (is_rootpe()) print *,'sppt_tau = ',sppt_tau
-   if (is_rootpe()) print *,'sppt = ',sppt
-   if (is_rootpe()) print *,'spptint = ',spptint
-   if (n_var_lndp>0) nlndp=1
-   if (n_var_spp>0) nspp=n_var_spp
-   if (is_rootpe())  print *,' nlndp   = ', nlndp
-   if (is_rootpe())  print *,' nspp   = ', nspp
-
-   if (nsppt > 0) allocate(rpattern_sppt(nsppt))
-   if (nshum > 0) allocate(rpattern_shum(nshum))
-   if (nskeb > 0) allocate(rpattern_skeb(nskeb))
+      endif
+    enddo
+
+    if (nsppt == 0) then
+      do_sppt = .false.
```

Comment:
Does "do_sppt" need to be changed to false here (above) if the run is going to be killed below with a return command anyway?

Reply:
Or, is this just setting "do_sppt" to false and then continuing the simulation (not killing it)?